lib/rules/no-restricted-properties.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * @fileoverview Rule to disallow certain object properties
+ * @author Will Klein
+ * @copyright 2016 Will Klein. All rights reserved.
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function(context) {
+    if (context.options.length === 0) {
+        return {};
+    }
+
+    var restrictedProperties = context.options.reduce(function(restrictions, option) {
+        var objectName = option.object;
+        var propertyName = option.property;
+
+        restrictions[objectName] = restrictions[objectName] || {};
+        restrictions[objectName][propertyName] = {
+            message: option.message
+        };
+
+        return restrictions;
+    }, {});
+
+    return {
+        "MemberExpression": function(node) {
+            var objectName = node.object && node.object.name;
+            var propertyName = node.property && node.property.name;
+            var matchedObject = restrictedProperties[objectName];
+            var matchedObjectProperty = matchedObject && matchedObject[propertyName];
+
+            if (matchedObjectProperty) {
+                var message = matchedObjectProperty.message ? " " + matchedObjectProperty.message : "";
+
+                context.report(node, "'{{objectName}}.{{propertyName}}' is restricted from being used.{{message}}", {
```

Comment:
Can you format as "Unexpected use of '{{objectName}}.{{propertyName}}.{{message}}"? (Fits more with our existing error messages.)